src/distributions.jl

```diff
@@ -22,7 +22,15 @@ function default_range(m::Distributions.MixtureModel, alpha = 0.0001)
 end
 
 yz_args(dist) = default_range(dist)
-yz_args(dist::Distribution{N, T}) where N where T<:Discrete = (UnitRange(Int.(default_range(dist))...),)
+function yz_args(dist::Distribution{N, T}) where N where T<:Discrete
+    minval, maxval = extrema(dist)
+    if isfinite(minval) && isfinite(maxval)  # bounded
+        sup = support(dist)
+        return sup isa AbstractVector ? (sup,) : ([sup...],)
```

Comment:
I guess it doesn't matter since I assume it only handles `Dirac` but one could avoid splatting with
```suggestion
        return sup isa AbstractVector ? (sup,) : (reduce(vcat, sup),)
```